File: DiscoDruid3/app/src/main/java/com/andruid/magic/discodruid/viewmodel/TrackViewModel.java
package com.andruid.magic.discodruid.viewmodel;

import android.app.Application;
import android.os.Bundle;
import android.support.v4.media.MediaBrowserCompat;

import com.andruid.magic.discodruid.data.Constants;
import com.andruid.magic.discodruid.datasourcefactory.TrackDataSourceFactory;
import com.andruid.magic.discodruid.model.Track;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.paging.LivePagedListBuilder;
import androidx.paging.PagedList;

public class TrackViewModel extends AndroidViewModel {
    private Application application;

    public TrackViewModel(@NonNull Application application) {
        super(application);
        this.application = application;
    }

    public LiveData<PagedList<Track>> getTracks(MediaBrowserCompat mediaBrowserCompat, Bundle options){
        TrackDataSourceFactory dataSourceFactory = new TrackDataSourceFactory(application.getApplicationContext(),mediaBrowserCompat, options);
        PagedList.Config config = new PagedList.Config.Builder()
                .setEnablePlaceholders(false)
                .setPageSize(Constants.PAGE_SIZE)
                .build();
        return new LivePagedListBuilder<>(dataSourceFactory,config).build();
    }
}

File: OldAttemptDiscoDruid/app/src/main/java/com/andruid/magic/discodruid/ui/viewmodel/TrackViewModel.kt
package com.andruid.magic.discodruid.ui.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList
import com.andruid.magic.discodruid.paging.track.TrackDataSourceFactory
import com.andruid.magic.medialoader.model.Track

class TrackViewModel : ViewModel() {
    companion object {
        private const val TRACK_PAGE_SIZE = 10
    }

    val tracksLiveData: LiveData<PagedList<Track>>

    init {
        val config = PagedList.Config.Builder()
            .setPageSize(TRACK_PAGE_SIZE)
            .setEnablePlaceholders(false)
            .build()
        tracksLiveData = LivePagedListBuilder<Int, Track>(TrackDataSourceFactory(), config)
            .build()
    }
}